src/main.rs

```diff
@@ -53,31 +56,31 @@ struct Opt {
     /// Dump the DMI data to a binary file
     #[structopt(long = "dump-bin", parse(from_os_str))]
     output: Option<PathBuf>,
+
+    /// Only display the entries of given type
+    #[structopt(short = "t", long = "type")]
+    bios_types: Option<Vec<BiosType>>,
```

Comment:
Would it possible to get access to multiple types?
Other than this question , it looks good to me.